tests/utils.ts

```diff
@@ -0,0 +1,189 @@
+import { UTXOType } from "../src";
+import * as crypto from 'crypto';
+import { Buffer } from 'buffer';
+
+// The following utilities are provided to determine the UTXOType of a transaction input.
+// This is necessary for parsing the test vectors from BIP352, but in practice a sending
+// wallet will already know the UTXOType for each UTXO it indends to spend and can set the
+// UTXOType field directly.
```

Comment:
how complete (albeit inefficient) this code is? is it worth exporting it and makign it apart of the lib so poor souls who implement wallets but dont know their input types can throw it in and use it when  using this js lib?

Reply:
Hm, not sure it makes sense for a sending wallet. A sending wallet will always know what kind of UTXOs it has based on what kind of addresses the wallet hands out, i.e. a wallet that only hands out native segwit and taproot addresses will only have native segwit and taproot UTXOs. It also needs to know the UTXO type to know how sign for it.

Also, for a sending wallet to use this function, they would need extra data that I'm pretty sure they won't have (like the prevout scriptPubKey).